crates/threshold-signature-server/src/validator/api.rs

```diff
@@ -273,8 +274,8 @@ pub async fn validate_new_reshare(
         .await?
         .ok_or_else(|| ValidatorErr::ChainFetch("Not Currently in a reshare"))?;
 
-    if reshare_data.new_signer != chain_data.new_signer
-        || chain_data.block_number != reshare_data.block_number
+    if chain_data.new_signers != reshare_data.new_signers
+        || chain_data.block_number != reshare_data.block_number.saturating_sub(1)
```

Comment:
Comparing the two `new_signers` collections is probably more expensive than the block number check, so it's faster to flip them around.
```suggestion
    if chain_data.block_number != reshare_data.block_number.saturating_sub(1)
        || chain_data.new_signers != reshare_data.new_signers
```